internal/kafka/internal/handlers/cluster.go

```diff
@@ -28,23 +31,43 @@ func (h clusterHandler) RegisterEnterpriseCluster(w http.ResponseWriter, r *http
 
 	ctx := r.Context()
 
+	provider, err := h.providerFactory.GetProvider(api.ClusterProviderOCM)
+
+	validateProviderError := func() handlers.Validate {
```

Comment:
Can we refactor this to its own method?

Reply:
moved

Reply:
Can we scope it further?

The change that has been made is create a function called `validateProvider` in the `internal/kafka/internal/handlers/validation.go` file.

That comes up with a set of issues:
1. It is a global-package variable: All files in that package will have access to that function. This pollutes the package namespace as I don't think all the code in the package needs to be aware of it 
2. As I see `internal/kafka/internal/handlers/validation.go` there should go validation functions that are useful to be used by multiple handlers. At the moment I don't think that's the case. Even if it were reusable it might make sense only in one specific handler which to me would indicate it wouldn't belong to that file either
3. Given that the function is a function at package level in a quite generic package name, it is not clear what `Provider` means in that context. It validates a "provider". What is a provider?

I think we should limit the scope of the validators as much as possible.

Reply:
@miguelsorianod this has been addressed, correct?

Reply:
Yes it's been addressed :+1:

Reply:
Thanks